datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneFineGrainDataMap.java

```diff
@@ -182,7 +182,7 @@ private String getQueryString(Expression expression) {
     // execute index search
     TopDocs result;
     try {
-      result = indexSearcher.search(query, indexReader.maxDoc());
+      result = indexSearcher.search(query, Integer.MAX_VALUE);
```

Comment:
I think it is better to pass this variable from the TEXT_MATCH UDF, like TEXT_MATCH("name:a", 100), so that user can decide the value. If user does not give the value, default value can be Integer.MAX_VALUE

Reply:
Added

Reply:
@Indhumathi27 i think what jacky means is, when user fires sql like "select * from table wjere text_match('name:a',100)", he should fire with number of records in result from lucene search, if he doesnot give in sql, then we will pass integer max , so please handle according to that in udf

Reply:
@jackylk Tired to  implement Text_Match UDF with variable number of arguments(either 1 0r 2).but it doesnt work.Only either one works

Reply:
@jackylk Implemented new function TEXT_MATCH_WITH_LIMIT to support user to fire query with number of records.TEXT_MATCH will return Integer.MaxValue number of records.